core/modules/file.py

```diff
@@ -544,25 +555,25 @@ def getUploadURL(self, fileName, mimeType, size=None, skey=None, *args, **kwargs
             else:
                 if not self.canAdd("leaf", None):
                     raise errors.Forbidden()
+
             maxSize = None  # The user has some file/add permissions, don't restrict fileSize
+
         if maxSize:
-            try:
-                size = int(size)
-                assert size <= maxSize
-            except:  # We have a size-limit set - but no size supplied
-                raise errors.PreconditionFailed()
+            if size > maxSize:
+                raise errors.Forbidden("Size limit exceeds maximum size")
```

Comment:
Can this cause an error if no size is specified?

Reply:
I am not entirely sure, but it seems to me that size is required. So if not specified, it should cause an error. Size needs to be exact according to initializeUpload doc str. It gets passed to the blob API in order to create a resumable upload session.

Reply:
> Can this cause an error if no size is specified?

Yup, if the size isn't provides as kwarg, it's None. Even with the type hint. 
Therefore the case `size is None` is missing.

Reply:
Maybe we ca use something like
```suggestion
            try:
                size = int(size)
            except:
                raise errors.NotAcceptable(f"Invalid size {size} provided")
            if size > maxSize:
                raise errors.Forbidden("Size limit exceeds maximum size")
``` 
I don't know a better solution like `errors.NotAcceptable`

Reply:
Ok, why not turn the size-parameter by default to 0 instead of None? As it's type-hinted with `int` it should be rejected in case it doesn't contain a valid int.

Reply:
> Ok, why not turn the size-parameter by default to 0 instead of None? As it's type-hinted with `int` it should be rejected in case it doesn't contain a valid int.

I would prefer `-1`. An empty file (`size = 0`) would be theoretically possible. Even though it probably makes no sense. But a negative value clearly indicates a different meaning.